pil-analyzer/tests/condenser.rs

```diff
@@ -0,0 +1,49 @@
+use powdr_number::GoldilocksField;
+use powdr_pil_analyzer::analyze_string;
+use test_log::test;
+
+use pretty_assertions::assert_eq;
+
+#[test]
+fn new_witness_column() {
+    let input = r#"namespace N(16);
+    let new_wit = || { let x; x };
```

Comment:
Can you also add some tests where this function adds constraints over the new column?

Reply:
Ah nice, that actually found a bug.

Reply:
I meant a case that adds constraints to the new column inside the function itself, like
```
let new_wit = || { let x; x = 2; x }
```

Or is that not possible atm?

Reply:
constraints can only be added in #1187